app/src/main/java/org/mozilla/firefox/vpn/device/ui/DevicesAdapter.kt

```diff
@@ -145,34 +137,35 @@ class DevicesAdapter(
 
     class LimitReachHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
 
-    class DevicesViewHolder(
-        override val containerView: View
-    ) : RecyclerView.ViewHolder(containerView), LayoutContainer {
+    class DevicesViewHolder(itemView: View, private val deleteListener: (DevicesViewHolder) -> Unit) : RecyclerView.ViewHolder(itemView) {
+
+        private val binding = ItemDeviceBinding.bind(itemView)
 
         fun bind(deviceUiMode: DeviceItemUiModel, currentDevice: CurrentDevice?) {
             val device = deviceUiMode.info
-            title.text = device.name
-            time.text = getRelativeTime(device.createdAt)
+            binding.title.text = device.name
+            binding.time.text = getRelativeTime(device.createdAt)
+            binding.delete.setOnClickListener { deleteListener(this) }
```

Comment:
Since deleteListener and ViewHolder is 1 to 1 related, I think this can be moved to others place like constructor instead of in the bind()